aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js

```diff
@@ -34,21 +40,66 @@ define(["dojo/_base/declare",
        *
        * @instance
        */
-      setupServer: function alfresco_testing_MockXhr__setupServer() {
+      setupServer: function alfresco_testing_SearchMockXhr__setupServer() {
          try
          {
-            var templateToUse = xhrSearchResponse;
             this.server.respondWith("GET",
-                                    /(.*)/,
-                                    [200,
-                                     {"Content-Type":"application/json;charset=UTF-8"},
-                                     templateToUse]);
+                                    /(.*)proxy\/alfresco\/slingshot\/search(.*)/,
+                                    lang.hitch(this, this.onSearch));
+
+            this.server.respondWith("GET",
+                                    /(.*)proxy\/alfresco\/slingshot\/auto-suggest(.*)/,
+                                    lang.hitch(this, this.onAutoSuggest));
             this.alfPublish("ALF_MOCK_XHR_SERVICE_READY");
          }
          catch(e)
          {
             this.alfLog("error", "The following error occurred setting up the mock server", e);
          }
+      },
+
+      /**
+       * @instance
+       */
+      onSearch: function alfresco_testing_SearchMockXhr__onSearch(request) {
+         var url = request.url;
+         var query = url.substring(url.indexOf("?") + 1);
+         var queryObj = ioQuery.queryToObject(query);
+         if (queryObj.term === "hame" && queryObj.spellcheck === "false")
+         {
+            request.respond(200, {
+               "Content-Type": "application/json;charset=UTF-8"
+            }, original);
+         }
+         else if (queryObj.term === "hame" && queryObj.spellcheck === "true")
```

Comment:
Lol... no, that's the point of the test - it's testing that alternative search suggestions are provided when you type something incorrectly, e.g. you enter "hame" but it automatically searches for "home" and tells you - but you can tell it to search for "hame" if you really meant it!